system/diagnostic_graph_aggregator/src/common/graph/config.cpp

```diff
@@ -264,9 +264,9 @@ void topological_sort(FileConfig & config)
   for (const auto & unit : config.units) units.push_back(unit.get());
 
   // Count degrees of each unit.
-  for (const auto & unit : units) {
-    if (const auto & link = unit->item) ++degrees[link->child];
-    for (const auto & link : unit->list) ++degrees[link->child];
+  for (const auto * const unit : units) {
+    if (const auto * const link = unit->item) ++degrees[link->child];
```

Comment:
@isamu-takagi 
Is it a correct implementation?

Reply:
Yes. The `item` is not a vector, so it is correct to use `if`.
https://github.com/autowarefoundation/autoware.universe/blob/main/system/diagnostic_graph_aggregator/src/common/graph/config.hpp#L55-L56

Reply:
@isamu-takagi 
Understood, thank you!!